addons/hr/models/hr_employee.py

```diff
@@ -359,9 +359,20 @@ def _onchange_timezone(self):
         if self.resource_calendar_id and not self.tz:
             self.tz = self.resource_calendar_id.tz
 
+    def _remove_work_contact_id(self, user):
+        """ Remove work_contact_id for previous employee if the user is assigned to a new employee """
+        partner_employee_ids = user.partner_id.employee_ids
+        # For employees with a user_id, the constraint (user can't be linked to multiple employees) is triggered
+        old_partner_employee_ids = partner_employee_ids.filtered(lambda e:
+            not e.user_id
+            and e.company_id == self.env.company
```

Comment:
Not sure about that :thinking: When ticking multiple companies from the selector, can't you create employees for a company that's not the "main" one ? Also, how does it work for branches ? :thinking:

Reply:
Indeed, the approach should be to check the current employee's company and filter only employees in that company. This way, we don't remove the work_contact_id of employees in other companies. I tested multiple companies selected and branches, and in both cases, it works as intended. The link between a partner and an employee is only removed _within the company of the employee_ and not other companies, if the user is assigned to a new employee.